src/solids4FoamModels/numerics/newGGIInterpolation/newGGIInterpolationPolygonIntersection.C

```diff
@@ -333,6 +341,264 @@ scalar newGGIInterpolation<MasterPatch, SlavePatch>::area2D
 }
 
 
+template<class MasterPatch, class SlavePatch>
+Tuple2<scalar,scalar>
+newGGIInterpolation<MasterPatch, SlavePatch>::normalGapIntegration
+(
+    const List<point2D>& clippedPolygon,
+    const orthoNormalBasis& orthoBase,
+    const label& faceMi,
+    const label& faceSi
+) const
+{
+    // Preliminaries
+    scalar penArea = 0.0;
+    const vector& slaveFaceNormal = slavePatch_.faceNormals()[faceSi];
+    const vector& masterFaceNormal = masterPatch_.faceNormals()[faceMi];
+    const pointField& slavePatchPoints = slavePatch_.localPoints();
+    const pointField& masterPatchPoints = masterPatch_.localPoints();
+
+    const vector& slaveFaceCentre =
+        slavePatch_[faceSi].centre(slavePatchPoints);
+
+    // Master face center is offset for ortho base c.s.
+    const vector& orthoBaseOffset =
```

Comment:
Same as above. This should not be a reference as `face::centre(const pointField&) const` returns a copy of a point (defined in face.H as far as I can see), i.e. switch this to `const vector orthoBaseOffset =`.